lib/index.js

```diff
@@ -67,10 +88,11 @@ async function batchRequest(batchSize, pageSize, requestParams={}, method, schoo
  * @class Class to interact with the Server
  */
 
-class Banner {
+class Banner { // Class makes it easier more readbale imo
```

Comment:
Yes, but the functions should not be directly in the class, otherwise you could call a method for which there is not request in Banner for that school, that is the entire purpose of the factory being used in the constructor. Please consider moving the class declaration and constructor back into their own file and importing the functions as was done before.